CCompiler/Token.cs

```diff
@@ -5,17 +5,26 @@ public class Token
     public readonly TokenType type;
     public readonly string text;
 
+    /// <summary>
+    /// origin index in file , do not have this in Equals
+    /// </summary>
+    public readonly int fileIndex;
+
     public Token(string text, TokenType type)
+      : this(text, type, -1)
```

Comment:
Is -1 a magic value, denoting the entire file?

Reply:
It's changed
I am preparing for line number and file but not fully implemented ,
uses line number now instead of position